### ISSUE TITLE
feat(compute): prefix-sum interval scoring kernel to replace the [L,L] span matrix

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#13272]
# feat(compute): prefix-sum interval scoring kernel to replace the [L,L] span matrix

```routing
lane: compute
paths: ["internal/compute/intervalscore.go", "internal/compute/intervalscore_test.go"]
expected_steps: 5
cross_key: study/gliner2-b1-prefixsum-interval-scoring
```

## Working spine
An interval-scoring kernel that scores every (start, end) interval from a single per-token
cumulative prefix, instead of materializing an L x L score matrix.

## Current state
Long-context candidate scoring in the fak native path is O(L^2) in the *materialized* tensor: an
`[L, L]` score matrix is either built directly or avoided ad hoc per call site. There is no shared
prefix-sum interval kernel, so every new interval-scoring consumer re-invents (or accidentally
re-introduces) the quadratic tensor. The study of `fastino-ai/GLiNER2` at pin
`d7c727458bf6929bc9ef5ee04e13c3f717a7c455` recorded a concrete, licensed (Apache-2.0) reference for
the linear-memory form: score each token's "inside" contribution once, take a `cumsum` prefix of
length `L + 1`, and recover an interval score as `prefix[end] - prefix[start]`
(`gliner2/models/boundary/heads.py:88-103`, consumed at `gliner2/models/boundary/scoring.py:253-255`).
Their implementation also centers each query before the fp32 cumulative sum and carries the mean
separately, which bounds the running offset while preserving exact interval differences.

## Why this is next
Reconnaissance says the capability is ABSENT in both repos: `fak-dev feature query "schema-conditioned
span extraction constrained decoding"` returns only grammar-constrained tool-call decoding notes.
Interval scoring is the memory-bounding primitive that any future long-context span/candidate-ranking
work needs, and it is independently witnessable today (no model weights, no new model architecture).

## Parent context
Study note `docs/research/STUDY-GLINER2-2026-09-18.md` in the private research
registry (borrow B1).

## Core through-line
Add `internal/compute/intervalscore.go`: a function that takes per-token inside logits `[..., L]` and
returns the `[..., L+1]` prefix plus a mean term, and a function that returns interval scores from
`(start, end)` index pairs. The change reaches a real seam: a single shared kernel that existing and
future interval-scoring consumers call, with an observable outcome — peak materialized tensor drops
from O(L^2) to O(L) for a fixed L, verified by a test that asserts the largest intermediate is
`L + 1` long — and a witness: a table-vs-prefix equivalence test plus a memory-bound assertion.

## Gold-plating boundary
Out of scope: adding a new model architecture, porting GLiNER2's encoder or any checkpoint, changing
existing attention implementations, and any GPU kernel authoring (SIMD/ROCm/Vulkan) in this leaf. This
issue lands the CPU reference kernel and its equivalence witness only.

## Done condition / witness
The prefix-sum interval kernel exists, is used by at least one real call site (or, if no consumer
exists yet, is proven equivalent to the explicit per-interval summation it replaces), and the package
test suite is green.

Witness: `go -C C:/work/fak test -count=1 ./internal/compute/... -run TestIntervalScoringPrefixEquivalence`

## Witness
Witness: `go -C C:/work/fak test -count=1 ./internal/compute/... -run TestIntervalScoringPrefixEquivalence`

## Definition of done
- [ ] `internal/compute/intervalscore.go` implements prefix construction (centered fp32 cumsum + carried mean) and interval retrieval.
- [ ] Golden/table equivalence test proves prefix interval scores equal explicit per-interval summation within fp tolerance.
- [ ] A memory-bound assertion proves the largest intermediate is `L + 1` long, not `L * L`.
- [ ] A caller in the tree uses the kernel, or the issue records why no consumer exists yet (presence-vs-invokability honesty).
- [ ] `go test ./internal/compute/...` is green and `gofmt`/`go vet` are clean.

## Acceptance gate
`go -C C:/work/fak test -count=1 ./internal/compute/...` exits 0 with the new tests included.

## Value frame
- Centrality: Core
- P1 Context: advanced - a shared memory-bounding primitive for long-context interval scoring
- P2 Net value: advanced - peak materialized tensor drops from O(L^2) to O(L) for interval scoring
- P3 Adaptation: N/A - additive kernel, no existing interface changed
- P4 Operations: advanced - real path, additive and independently witnessable without model weights

## Closure binding
The resolving commit cites this issue (`Fixes #<num>`) and carries the `(fak intervalscore)` trailer.

## Lane
compute

## Likely files
- internal/compute/intervalscore.go
- internal/compute/intervalscore_test.go

## Expected steps
5

- Centrality: Core
- P1 Context: advanced - shared primitive that bounds long-context scoring memory
- P2 Net value: advanced - O(L^2) to O(L) materialized tensor for interval scoring
- P3 Adaptation: N/A - additive kernel, no existing interface changed
- P4 Operations: advanced - real path, additive and independently witnessable

## Work estimate
Estimate: 3 points

## Overall completion contribution
Contribution: 3/8 points

## Completion standard
Landed on main with the equivalence witness green.

## Provenance
- Source study: `fastino-ai/GLiNER2` pin `d7c727458bf6929bc9ef5ee04e13c3f717a7c455`, borrow B1.
- License: Apache-2.0 (adapt permitted with attribution).
- Hardware disposition: source-only study; this issue's witness is CPU software-verified, NOT a physical-hardware claim.

**Comment by anthony-chaudhary** [comment]
Landed witness (auto-closed by the \Fixes #13272\ trailer). Commit \5adf5acae1a77bebcf2693a59a6967da27164003\ on public main (pushed \53933ba28..5adf5acae\). Witness: \go test ./internal/compute/ -run 'TestIntervalScoring' -count=1\ -> PASS (5 cases: prefix==explicit equivalence over every [start,end) span; O(L) memory bound L+1 not L*L; batch prefix reuse with malformed spans dropped; range refusals fail-closed; empty/single shapes). Full package ok (36.594s); \go vet\ clean; \gofmt -l\ clean; 5-gate \ak-boundary check --fak-dir <worktree>\ PASS (0 violations, 19033 files); leak audit exit 0. [SW-VERIFIED] CPU reference kernel; no physical-hardware criterion applies.
